### ISSUE TITLE
Downloads entire history

### ISSUE THREAD

**Issue by TheOctonaut** [badger707/esb-smart-meter-reading-automation#1]
Not really an issue per se but a note (didn't see a way to leave a 'discussion' like in some other projects).

I'm not sure what "startDate" does - it always downloads every record ever. I thought this might be a bug, or that something had changed again, but as it turns out that's the behaviour on the ESBN HDF data page too. It just straight up ignores what you've asked and gives you everything ever.

I only need the first two results for the HA automation I'm making but genuinely can't get ESB to _not_ send me everything. Huh.

**Comment by badger707** [comment]
Yeah, I have noticed that too... Well, in my use case I'm ok to get full file every time, and if I need specific part of it I can always transform it in python... I have dropped my initial plan to use Influx + Grafana, does not fit my needs. Instead, I have modified python to act as a sensor with bunch of attributes and dynamic data range parsing by directly pushing billing period dates ranges from HA native dashboard.
- I'm working on updating instruction with more step by step guide, quick preview:

![image](https://github.com/badger707/esb-smart-meter-reading-automation/assets/77490076/e0659faf-cede-43d0-9d4d-e7561b6b95a2)

![image](https://github.com/badger707/esb-smart-meter-reading-automation/assets/77490076/60f61bf1-43d8-430e-99be-e2459762ecc6)

**Comment by vincentezw** [comment]
Thanks for fixing the script and sharing your experience ❤️ 

I'd be interested to hear how you've got this configured. I found that new data becomes available only once per day, for the entire previous day. So we're dealing with a chunk of historic data, and without either injecting the data straight into the HA database, or using a history helper module, I couldn't find a nice way to configure this as a sensor.

I have [forked](https://github.com/vincentezw/esb-smart-meter-influxdb) the script to export straight to InfluxDB so I can run a daily cron. I made some other improvements too such as a config file, structure and deduplicating before saving to the database. If you'd like to merge those changes I can open a PR.

**Comment by badger707** [comment]
@vincentezw nice work on influx integration! My initial approach/testing for influx integration was not as clean as you have it there, every day we learn something new 😄 

From my observations, there could be more than few days of refresh delay from EBS, at one point I saw EBS was behind 6 days...

Making this to work as sensor is tricky and requires some weird things to perform on HA installation. 
My HA is running as VM on ESXi hypervisor cluster (perks of working at virtualisation giant company), I am unsure if it will work/behave same in Docker env, not a fan of Docker - might need some testing but thats not a  priority at the moment.

I will need to explore HACS development nuances to wrap this (sensor) in simple package so other folks could install it in less painful way.

PR - yes, lets keep this option open/hot for future revisions.

**Comment by Ciaran97** [comment]
Any more progress on the updated guide? specifically looking at getting this data from InfluxDB to a HA sensor.

**Comment by freddo99** [comment]
> Yeah, I have noticed that too... Well, in my use case I'm ok to get full file every time, and if I need specific part of it I can always transform it in python... I have dropped my initial plan to use Influx + Grafana, does not fit my needs. Instead, I have modified python to act as a sensor with bunch of attributes and dynamic data range parsing by directly pushing billing period dates ranges from HA native dashboard.
> 
> * I'm working on updating instruction with more step by step guide, quick preview:
> 
> ![image](https://private-user-images.githubusercontent.com/77490076/279488122-e0659faf-cede-43d0-9d4d-e7561b6b95a2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE1MTc3MjUsIm5iZiI6MTczMTUxNzQyNSwicGF0aCI6Ii83NzQ5MDA3Ni8yNzk0ODgxMjItZTA2NTlmYWYtY2VkZS00M2QwLTlkNGQtZTc1NjFiNmI5NWEyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTEzVDE3MDM0NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQ3MTI1NTY4YWNmYjU3ZDg2NjU3ZTdjMjdjMjQyOTU2ZGQwZjY5ZjQwNzM3YzczZWFiMGUyMmE5NjZkZGIxMTImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Z4FnXZ1Pmt5LulOrOiZCqjs8H6hYR-3fZyzRYEKrfOg)
> 
> ![image](https://private-user-images.githubusercontent.com/77490076/279488052-60f61bf1-43d8-430e-99be-e2459762ecc6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE1MTc3MjUsIm5iZiI6MTczMTUxNzQyNSwicGF0aCI6Ii83NzQ5MDA3Ni8yNzk0ODgwNTItNjBmNjFiZjEtNDNkOC00MzBlLTk5YmUtZTI0NTk3NjJlY2M2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTEzVDE3MDM0NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTE0Y2FiMzQwMTBiZjllYmRkNTE3ZWEwZDJmN2MxOGFjNzllMmM2MTQzMjdmODk0YmIwNzg0OGNmMjVkYmMwZGYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.SBE7c40XwMzaoGx68tUcn61x9hinf6H3lQs1KybW2qI)

This looks great. Thank you for sharing your work. Would it be possible to offer a guide for how to integrate with HA and create the sensors? Thank you.

**Comment by badger707** [comment]
The method I used to create sensor with attributes no longer works after recent HA version update.
